file_dropzone.module

```diff
@@ -0,0 +1,290 @@
+            unset($_POST['files']);
+            foreach (array('name', 'type', 'tmp_name', 'error', 'size', 'orig_name') as $key) {
+              $_FILES['files'][$key][$file_upload_name] = '';
+              if ($key == 'error') {
+                $_FILES['files'][$key][$file_upload_name] = 4;
+              }
+              if ($key == 'size') {
+                $_FILES['files'][$key][$file_upload_name] = 0;
+              }
+            }
+          }
+        }
+      }
+
+      if ($attach_fid) {
+        $triggering_element_name = $file_upload_name . '_attach_button';
+
+        $_POST['_triggering_element_name'] = $triggering_element_name;
+        $_POST['_triggering_element_value'] = t('Attach');
+        $_POST['media'][$file_upload_name] = $attach_fid;
+      }
+    }
+
+    $_REQUEST = $_POST + $_GET;
+
+    // Update the form input with the new post values.
+    $form_state['input'] = $_POST;
+
+    // Process user input. $form and $form_state are modified in the process.
+    drupal_process_form($form['#form_id'], $form, $form_state);
+
+    $results[$index] = array(
+      'status' => 201,
+      'message' => 'OK',
+      'fid' => 0,
+    );
+
+    $messages = theme('status_messages', array('display' => 'error'));
+    if (!empty($messages)) {
+      $results[$index] = array(
+        'status' => 400,
+        'message' => $messages,
+        'fid' => -1,
+      );
+    }
+
+    $index++;
+  }
+
+  lock_release($lock_name);
+
+  // Retrieve the element to be rendered.
+  foreach ($form_parents as $parent) {
+    $form = $form[$parent];
+  }
+
+  // Add the special Ajax class if a new file was added.
+  if (isset($form['#file_upload_delta']) && $current_file_count < $form['#file_upload_delta']) {
+    $form[$current_file_count]['#attributes']['class'][] = 'ajax-new-content';
+  }
+  // Otherwise just add the new content class on a placeholder.
+  else {
+    $form['#suffix'] .= '<span class="ajax-new-content"></span>';
+  }
+
+  $output = theme('status_messages') . drupal_render($form);
+  $js = drupal_add_js();
+  $settings = call_user_func_array('array_merge_recursive', $js['settings']['data']);
+
+  $commands[] = ajax_command_replace(NULL, $output, $settings);
+  $commands[] = array( 'command' => 'dropzoneFiles', 'results' => $results );
+  return array('#type' => 'ajax', '#commands' => $commands);
+}
+
+function file_dropzone_error_exit($message, $status_header = 'HTTP/1.0 400 Bad request') {
+    header($status_header);
+    print strip_tags($message);
+    exit();
+}
+
+/**
+ * Process callback for the file managed field element.
+ */
+function file_dropzone_file_field_widget_process($element, &$form_state, $form) {
+  $field = field_widget_field($element, $form_state);
+
+  // Adjust the Ajax settings so that on upload and remove of any individual
+  // file, the entire group of file fields is updated together.
+  // Also add our file_dropzone menu callback.
+  // @todo Make cardinality 1 work.
```

Comment:
I think this should give a normal error for number of files exceeded, but I will check that. Good point.